src/ccover/ccover_bmc.cpp

```diff
@@ -0,0 +1,234 @@
+/*******************************************************************\
+
+Module: Bounded Model Checking for Test Suite Generation
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+/// \file
+/// Bounded Model Checking for Test Suite Generation
+
+#include "ccover_bmc.h"
+
+#include <chrono>
+#include <iostream>
+
+#include <util/exit_codes.h>
+#include <util/string2int.h>
+#include <util/string_utils.h>
+
+#include <langapi/language_util.h>
+
+#include <goto-symex/memory_model_pso.h>
+#include <goto-symex/slice.h>
+#include <goto-symex/slice_by_trace.h>
+
+#include <cbmc/cbmc_solvers.h>
+
+#include <linking/static_lifetime_init.h>
+
+void ccover_bmct::do_conversion()
```

Comment:
Ideally, there should be a generic BMC class that provides these building block to avoid duplication.

Reply:
I don't see how this can be done in <2 years